hashes/src/siphash24.rs

```diff
@@ -226,6 +226,15 @@ impl Hash {
     /// Creates a hash from its (little endian) 64-bit integer representation.
     pub fn from_u64(hash: u64) -> Hash { Hash(hash.to_le_bytes()) }
 
+    /// Returns the underlying byte array.
+    pub const fn to_byte_array(self) -> [u8; 8] { self.0 }
+
+    /// Returns a reference to the underlying byte array.
+    pub const fn as_byte_array(&self) -> &[u8; 8] { &self.0 }
+
+    /// Constructs a hash from the underlying byte array.
+    pub const fn from_byte_array(bytes: [u8; 8]) -> Self { Self(bytes) }
+
```

Comment:
I believe we didn't miss Siphash, I think it broke when removing the `Default` bound since it was no longer implemented using macros but manually. We should really fix that by splitting the macro as I originally suggested sine then anything else we missed will get fixed out of the box.